### PR TITLE
documenting release process

### PR DESCRIPTION
**Pull request by feministy**
refs #4135 

I took the comments from @segiddins, @lynnco, and @indirect and organized them with a little extra prose where I saw fit. It's mostly y'alls words, just reorganized 😊 

There are a few questions I have, mostly about branching:

- When `1.12.pre.1` is released, is it merged to `1-12-stable`?
- If there are bugs in `1.12.pre.1` *and* the branch (`1-12-pre-1`) is merged to master, is the branch for `1.12.pre.2` a branch off of `1-12-stable`?
- Are there any places we can reference to follow on discussions about in progress features, roadmaps, planned work, etc? It might be a good place to reference some of that stuff
- How can people get in touch with the core team?
- Is there a template for blog posts?

A few things to note:

- In the notes on the issue, it looks like you flop between using `.` versus `-` as the delimiter in your branch names. I went with `-` since that appears to be what you're using in the repo itself
- I do the full Github links so when people pull it down locally, they have the reference to the full link locally instead of relying on Github's magic linking

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+At this point, you're a release manager! Pour yourself several tasty drinks and
+think about taking a vacation in the tropics.
+
+#### Branching
+
+Minor releases of the next version start with a new release branch from the
+current state of master: `1-12-stable`
+
+From that release branch, we create the first pre-release branch for that minor
+version: `1-12-0-pre-1`. Until `1-12-0-pre-1` is ready for release, all active
```
Comment (segiddins) [inline RELEASING.md]: we only do branches for the minor series -- there is no `1-12-0` branch, for example

**Comment by colby-swandale** [comment]
ping @feministy

**Comment by indirect** [comment]
> When 1.12.pre.1 is released, is it merged to 1-12-stable?

We generally branch off `1-12-stable` from master first, and then release version `1.12.pre.1` from the tip of the `1-12-stable` branch.

> If there are bugs in 1.12.pre.1 and the branch (1-12-pre-1) is merged to master, is the branch for 1.12.pre.2 a branch off of 1-12-stable?

Since there is only one branch per minor version (`1-11-stable` and then `1-12-stable`), we put all changes to the 1.12.x series of releases into the `1-12-stable` branch. After doing a release, we merge the exact commit that was released into master to make sure that it's up to date.

> Are there any places we can reference to follow on discussions about in progress features, roadmaps, planned work, etc? It might be a good place to reference some of that stuff

We generally use a GitHub issues milestone to track releases, but we don't have a lot of formal tracking: https://github.com/bundler/bundler/milestones

> How can people get in touch with the core team?

Slack is easiest, but we also have a shared email address: team@bundler.io.

> Is there a template for blog posts?

Not yet, but I started working on something to automate new releases on the website, and it has a template for the "what's new" page in it: https://github.com/bundler/bundler-site/pull/307

**Comment by schneems** [comment]
What’s the status here. Is this PR still being worked on?

**Comment by feministy** [comment]
@schneems I have not had time to devote to this PR, everything I have pushed up so far is all of the work I've done. If someone wants to take it over, that's ok with me!

**Comment by segiddins** [comment]
I've just updated this, I'd greatly appreciate some review ❤️

File: doc/development/RELEASING.md [hunk 2]
```diff
@@ -1,9 +1,147 @@
 # Releasing
 
-_Release process documentation is in progress, see [PR 5252](https://github.com/bundler/bundler/pull/5252)._
+Bundler users [Semantic Versioning](https://semver.org/).
```
Comment (segiddins) [inline doc/development/RELEASING.md]: `uses`

File: doc/development/RELEASING.md [hunk 3]
```diff
@@ -1,9 +1,147 @@
...
+## Changelog
+
+Bundler maintains a list of changes present in each version in the `CHANGELOG.md` file.
+Entries should not be added in pull requests, but are rather written by the Bundler
+maintainers in the [bundler-changelog repo](https://github.com/bundler/bundler-changelog).
+That reposity tracks changes by pull requests, with each entry having an associated version,
+PR, section, author(s), issue(s) closed, and message.
+
+Ensure that repo has been updated with all new PRs before releasing a new version,
+and copy over the new sections to the `CHANGELOG.md` in the Bundler repo.
```
Comment (indirect) [inline doc/development/RELEASING.md]: How does one update that repo? I honestly don't remember, and that repo doesn't have a readme. 😭

Comment (segiddins) [reply doc/development/RELEASING.md]: uh, I haven't done so since my last release. want to open an issue there for someone to add a README ?

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**